scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -177,9 +182,13 @@ object Type {
   }
   @ast class Apply(tpe: Type, args: List[Type] @nonEmpty) extends Type
   @ast class ApplyInfix(lhs: Type, op: Name, rhs: Type) extends Type
-  @ast class Function(params: List[Type], res: Type) extends Type
+  @branch trait FunctionLike extends Type {
```

Comment:
Not super keen on the names, but I have no idea what can be better :sweat_smile:

Reply:
Maybe FunctionTree and CaseTree ?

Reply:
> Maybe FunctionTree and CaseTree ?

that might work.

other possibilities:
- `FooLike` (for completeness; this is what you used, some scala libraries do that)
- `FooBase`
- `AFoo`
- `FooBar` if it `extends Bar` (so for this line, `FunctionType` rather than `FunctionTree`)

Reply:
Let's do CaseTree, FunctionTerm and FunctionType. I fell best with that.